protocol/codecs/base_scalar.go

```diff
@@ -137,18 +137,15 @@ func (c *UUID) Type() reflect.Type {
 }
 
 // Decode a UUID.
-func (c *UUID) Decode(bts *[]byte, out reflect.Value) {
-	protocol.PopUint32(bts) // data length
+func (c *UUID) Decode(msg *buff.Message, out reflect.Value) {
 	p := (*types.UUID)(unsafe.Pointer(out.UnsafeAddr()))
-	copy((*p)[:], (*bts)[:16])
-	*bts = (*bts)[16:]
+	copy((*p)[:], msg.PopBytes())
```

Comment:
I'm just curious: why do we do unsafe pointer stuff here and don't do it in `(c *Bytes) Decode()`?

Reply:
If we did that the buffer could not be reused because the memory would be shared with the calling code.

Reply:
I think I misunderstood the question. I intentionally kept this change to the buffer. I plan to rip out reflect.Value from the Decoder.Decode() call and replace it with unsafe.Pointer every where. This should be a significant speed up.